### PR TITLE
Localstorage for form

### PR DESCRIPTION
**Pull request by Itswali**
Add local storage to save data of form.

### CODE REVIEW

**Review by topeogunleye — COMMENTED** [review]
> Hi @Itswali,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
## Invalid Code Review Request🛑 

>- [ ] Kindly update your linters.yml file for this project to include ESLinters. In the root project, you have a folder named .github. Inside this one, you have another folder named workflows. And inside the workflows folder, you have a file named linters.yml. Inside this file, you need to update the linters. The linters needed for this project can be found [here](https://github.com/microverseinc/linters-config/tree/master/html-css-js);

image

> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

File: main.js [hunk 1]
```diff
@@ -307,4 +307,14 @@ form.addEventListener('submit', (e) => {
...
+
+form.addEventListener('input', () => {
+  const contactForm = {
+    name: userName.value,
+    email: userEmail.value,
+    message: userMsg.value,
+  };
+
+  localStorage.setItem('inputForm', JSON.stringify(contactForm));
 });
```
Comment (RolandM99) [inline main.js]: - [ ] In order to be used after being retrieved from the local storage the saved string should be turned back into an object using `JSON.parse` or similar methods. take a look here:
```javascript
let savedFormData = localStorage.getItem('contactForm');
savedFormData = JSON.parse(savedFormData);
```
- [ ] Also, When the user loads the page, the code should checks if there are any saved data in the local storage, and if so the input fields should be pre-filled with the data. you can take a look at the code snippet below. NB: this is just an example of how you can implement it.
```javascript
window.onload = () => {
  let savedFormData = localStorage.getItem('contactForm');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: UserName.value = savedFormData.name
  }
};
```

**Review by RolandM99 — CHANGES_REQUESTED** [review]
## Hello Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 1 above)
Comment (Itswali) [reply main.js]: @RolandM99 can u please check I implemented your required changes.

**Review by codecaiine — APPROVED** [review]
Hi @Itswali , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good project implementation :heavy_check_mark: 
- Great required changes implementation :heavy_check_mark: 
- No linters errors :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**